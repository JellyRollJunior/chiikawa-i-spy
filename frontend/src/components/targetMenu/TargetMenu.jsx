import styles from './TargetMenu.module.css';
import shared from '../../styles/shared.module.css';

const TargetMenu = ({ targets, isVisible, x, y }) => {
  const style = { left: x, top: y };
  style.display = isVisible ? 'block' : 'none';

  return (
    <>
      <div className={styles.menuWrapper} style={style}>
        <div className={styles.menu}>
          <div className={styles.targetBox}></div>
          <ul className={styles.optionWrapper}>
            {targets && targets.map((target) =>
              <li>
                <button className={`${styles.menuOption} ${shared.primaryButton}`}>{target.name}</button>
              </li>
            )}
          </ul>
        </div>
      </div>
    </>
  );
};

export { TargetMenu };
