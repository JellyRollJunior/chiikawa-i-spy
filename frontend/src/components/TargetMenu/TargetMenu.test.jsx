import { vi, describe, it, expect } from 'vitest';
import { screen, render } from '@testing-library/react';
import { TargetMenu } from './TargetMenu.jsx';
import userEvent from '@testing-library/user-event';

const TARGETS = [
  {
    id: 1,
    name: 'Floppy-Eared Usagi',
    isFound: false,
  },
  {
    id: 2,
    name: 'Satapanbin Shisa',
    isFound: false,
  },
];

describe('Target menu component', () => {
  it('renders target box and targets when visible', () => {
    const { container } = render(<TargetMenu targets={TARGETS} isVisible={true} />);

    expect(container).toMatchSnapshot();
  });

  it('only renders targets that are not found', () => {
    const TARGETS = [
        {
            id: 1,
            name: 'Floppy-Eared Usagi',
            isFound: true,
        },
        {
            id: 2,
            name: 'Satapanbin Shisa',
            isFound: false,
        },
    ];
    render(<TargetMenu targets={TARGETS} isVisible={true} />);

    // only one button should be rendered with name Satapanbin Shisa
    expect(screen.getByRole('button').textContent).toStrictEqual(TARGETS[1].name);
    expect(screen.getAllByRole('button').length).toBe(1);
  });

  it('Calls handle guess function when buttons are clicked', async () => {
    const handleGuess = vi.fn()
    render(<TargetMenu targets={TARGETS} isVisible={true} handleGuess={handleGuess} />);
    const user = userEvent.setup();

    await user.click(screen.getByRole('button', { name: TARGETS[0].name }));

    expect(handleGuess).toHaveBeenCalled();
    expect(handleGuess).toHaveBeenCalledWith(TARGETS[0].id)
  })
});
