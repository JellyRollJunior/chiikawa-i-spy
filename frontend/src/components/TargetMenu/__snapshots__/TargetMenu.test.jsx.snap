// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Target menu component > renders target box and targets when visible 1`] = `
<div>
  <div
    class="_fadeIn_5890fc _menuWrapper_845c23"
    style="left: 0px; top: 0px; display: block;"
  >
    <div
      class="_menu_845c23"
    >
      <div
        class="_targetBox_845c23"
      />
      <ul
        class="_optionWrapper_845c23"
      >
        <li>
          <button
            class="_menuOption_845c23 _primaryButton_5890fc"
          >
            Floppy-Eared Usagi
          </button>
        </li>
        <li>
          <button
            class="_menuOption_845c23 _primaryButton_5890fc"
          >
            Satapanbin Shisa
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
